src/Symfony/Component/Dsn/Factory/MemcachedConnectionFactory.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Dsn\Factory;
+
+use Symfony\Component\Dsn\Exception\InvalidArgumentException;
+
+/**
+ * Factory for Memcached connections.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class MemcachedConnectionFactory
```

Comment:
shouldn't they implement an interface?

Reply:
each factory has a different return type, so I see no point creating an interface (the return type would have to be `mixed`

Reply:
the input could also vary (`string` in redis, `array` or `string` in memchached).